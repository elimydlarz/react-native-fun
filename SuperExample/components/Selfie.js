import React, { Component } from 'react';
import { View, Image, Button } from 'react-native';

const COLOURS = {
  NEUTRAL: 'black',
  LIKED: 'green',
  DISLIKED: 'red',
};

export default class component extends Component {
  state = {
    vote: 'NEUTRAL',
  }

  render() {
    const { source } = this.props;
    const borderColor = COLOURS[this.state.vote];

    return (
      <View style={{ margin: 5 }}>
        <Image
          style={{ borderColor, borderWidth: 2, width: 100, height: 100 }}
          resizeMode="cover"
          source={source}
        />
        <View style={{ width: 100, flexDirection: 'row', justifyContent: 'space-between' }}>
          <Button
            color={COLOURS.LIKED}
            title="+"
            onPress={() => this.setState({ vote: 'LIKED' })}
          />
          <Button
            color={COLOURS.DISLIKED}
            title="-"
            onPress={() => this.setState({ vote: 'DISLIKED' })}
          />
        </View>
      </View>
    );
  }
}
