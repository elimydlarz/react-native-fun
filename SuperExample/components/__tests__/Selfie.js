import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Selfie from '../Selfie';

test('Selfie matches snapshot', () => {
  const tree = renderer.create(
    <Selfie source={{ uri: 'example' }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
