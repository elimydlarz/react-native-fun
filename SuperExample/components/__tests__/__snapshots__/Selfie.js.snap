// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selfie matches snapshot 1`] = `
<View
  style={
    Object {
      "margin": 5,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "example",
      }
    }
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 2,
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 100,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "green",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
  </View>
</View>
`;
